tendermint/src/lite_impl/signed_header.rs

```diff
@@ -71,15 +68,21 @@ impl lite::Commit for block::signed_header::SignedHeader {
                         if header_hash != self.header_hash() {
                             fail!(
                                 lite::error::Kind::ImplementationSpecific,
-                                "vote from validator {} is not for the current header",
-                                precommit.validator_address
+                                "validator({}) voted for header {}, but current header is {}",
+                                precommit.validator_address,
+                                header_hash,
+                                self.header_hash
                             );
                         }
                     }
                 }
                 None => (),
             }
         }
+
+        // TODO: check here if all the votes are from correct validators
+        // TODO: return error kind "InvalidValidator" if we find a vote from a validator not in the val set
```

Comment:
Are these tracked in issues too?

Reply:
Will update in issue #140 once the code is added. Do you think it's OK for now to just return an error instead of marking it as faulty?

Reply:
Awesome, thanks! Yes, I think that is OK for now.

Reply:
Great, Thanks :)